cpp/models/abm/location.h

```diff
@@ -260,6 +254,21 @@ class Location
         m_capacity_adapted_transmission_risk = consider_capacity;
     }
 
+    /**
+     * Store the current number of individuals in the each infection state at a specific time step
+     * @param t the time step to save
+    */
+    void store_current_population(const TimePoint& t);
+
+    /**
+     * Return the time series object of the current number of individuals in the each infection state
+     * @return the time series object of the current number of individuals in the each infection state
+    */
+    const TimeSeries<double>& get_population() const
```

Comment:
Could you maybe try to go over to ScalarType instead of double? We have defined this in memilio and this could then even be replace by single precision arithmetic easily.